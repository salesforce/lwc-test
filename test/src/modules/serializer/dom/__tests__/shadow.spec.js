/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

it('prints shadow tree', () => {
    const host = document.createElement('div');
    const root = host.attachShadow({ mode: 'open' });

    const child = document.createElement('span');
    child.textContent = 'In the shadow DOM';
    root.append(child);

    expect(host).toMatchSnapshot();
});

it('prints slotted content properly', () => {
    const host = document.createElement('div');
    const root = host.attachShadow({ mode: 'open' });

    const text = document.createTextNode('In the shadow DOM');
    const slot = document.createElement('slot');
    root.append(text, slot);

    const slotted = document.createElement('span');
    slotted.textContent = 'In the light DOM';
    host.append(slotted);

    expect(host).toMatchSnapshot();
});

it('prints shadow tree recursively', () => {
    const host = document.createElement('div');
    const root = host.attachShadow({ mode: 'open' });

    const topText = document.createTextNode('In the top shadow DOM');
    const innerHost = document.createElement('span');
    const innerRoot = innerHost.attachShadow({ mode: 'open' });
    root.append(topText, innerHost);

    const innerText = document.createTextNode('In the inner shadow DOM');
    innerRoot.append(innerText);

    expect(host).toMatchSnapshot();
});
