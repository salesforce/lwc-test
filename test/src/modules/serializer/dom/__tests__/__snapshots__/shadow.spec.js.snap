// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints shadow tree 1`] = `
<div>
  #shadow-root(open)
    <span>
      In the shadow DOM
    </span>
</div>
`;

exports[`prints shadow tree recursively 1`] = `
<div>
  #shadow-root(open)
    In the top shadow DOM
    <span>
      #shadow-root(open)
        In the inner shadow DOM
    </span>
</div>
`;

exports[`prints slotted content properly 1`] = `
<div>
  #shadow-root(open)
    In the shadow DOM
    <slot />
  <span>
    In the light DOM
  </span>
</div>
`;
